tests/string_test.py

```diff
@@ -295,25 +294,31 @@ def setUp(self):
         ArkoudaTest.setUp(self)
         base_words1 = ak.random_strings_uniform(1, 10, UNIQUE, characters='printable')
         base_words2 = ak.random_strings_lognormal(2, 0.25, UNIQUE, characters='printable')
-        gremlins = ak.array(['"', ' ', ''])
-        self.gremlins = gremlins
+        gremlins = np.array(['"', ' ', ''])
+        self.gremlins = ak.array(gremlins)
         self.base_words = ak.concatenate((base_words1, base_words2))
         self.np_base_words = np.hstack((base_words1.to_ndarray(), base_words2.to_ndarray()))
         choices = ak.randint(0, self.base_words.size, N)
         self.strings = self.base_words[choices]
         self.test_strings = self.strings.to_ndarray()
         self.cat = ak.Categorical(self.strings)
         x, w = tuple(zip(*Counter(''.join(self.base_words.to_ndarray())).items()))
-        self.delim =  np.random.choice(x, p=(np.array(w)/sum(w)))
+        self.delim = self._get_delimiter(x,w,gremlins)
         self.akset = set(ak.unique(self.strings).to_ndarray())
-        self.gremlins_base_words = base_words = ak.concatenate((base_words1, base_words2, gremlins))
-        self.gremlins_strings = ak.concatenate((base_words[choices], gremlins))
+        self.gremlins_base_words = ak.concatenate((self.base_words, self.gremlins))
+        self.gremlins_strings = ak.concatenate((self.base_words[choices], self.gremlins))
         self.gremlins_test_strings = self.gremlins_strings.to_ndarray()
         self.gremlins_cat = ak.Categorical(self.gremlins_strings)
 
+    def _get_delimiter(self,x : Tuple, w : Tuple, gremlins : np.ndarray) -> str:
+        delim = np.random.choice(x, p=(np.array(w)/sum(w)))
+        if delim in gremlins:
```

Comment:
Rather than requiring `delim` not to be in `gremlins`, we should just make sure it's not the empty string. The other two gremlins (`"` and ` `) are valid delimiters.

Reply:
@reuster986 I made the change you requested and test_ends_with is failing when the delimiter = " " (one space):

![image](https://user-images.githubusercontent.com/10785153/106740130-01167880-65e8-11eb-8e43-a397d14f9862.png)

Reply:
@reuster986 I see what you mean about invalid delimiters. Both the double quote and space work correctly as long as the last string in the Strings object does not end with the delimiter:

![image](https://user-images.githubusercontent.com/10785153/106743281-2b6a3500-65ec-11eb-89b3-bbcb03e71724.png)

And, if the delimiter is an empty string, it fails for all strings in the Strings object that are an empty string.

@reuster986 @mhmerrill In this PR I am fixing the manifestation, not the root problem, where the root problem is that the ends_with logic fails when the last string in the Strings object ends with the delimiter. All I am attempting to do with this PR is to ensure the CI workflow succeeds. Please confirm how you'd like to proceed with this PR.